src/taglibs/migrate/async-fragment-tag.js

```diff
@@ -0,0 +1,59 @@
+const migrateControlFlowDirectives = require("./control-flow-directives");
+const newTags = {
+    "async-fragment": "await",
+    "async-fragments": "await-reorderer",
+    "async-fragment-placeholder": "await-placeholder",
+    "async-fragment-timeout": "await-timeout",
+    "async-fragment-error": "await-error"
+};
+module.exports = function migrator(oldNode, context) {
+    const oldTag = oldNode.tagName;
+    const newTag = newTags[oldTag];
+    var provider;
+    var varName;
+    var argument;
+    const attributes = oldNode.attributes;
+    const builder = context.builder;
+    migrateControlFlowDirectives(oldNode, context);
+    oldNode.setTransformerApplied(migrateControlFlowDirectives);
+
+    context.deprecate(
+        `The "<${oldTag}>" tag is deprecated. Please use "<${newTag}>" instead. See: https://github.com/marko-js/marko/wiki/Deprecation:-var-assign-invoke-tags`
```

Comment:
We will need to update this error message and create a new wiki page here: https://github.com/marko-js/marko/wiki/Deprecation:-async-fragment

This should follow a similar format to https://github.com/marko-js/marko/wiki/Deprecation:-var-assign-invoke-tags